src/praisonai-mcp/praisonai_mcp/mcp_server/transports/stdio.py

```
@@ -48,71 +56,76 @@ async def run(self) -> None:
-            asyncio.streams.FlowControlMixin, sys.stdout
+        loop = asyncio.get_running_loop()
+        line_queue: "queue.Queue[Optional[bytes]]" = queue.Queue()
+
+        def _blocking_reader() -> None:
+            """Read stdin in a dedicated thread; ``None`` signals EOF."""
+            try:
+                for raw in sys.stdin.buffer:
+                    if not self._running:
+                        break
+                    line_queue.put(raw)
+            except Exception as exc:  # pragma: no cover - defensive
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)